main/cmds.go

```diff
@@ -197,28 +214,40 @@ func enable(ctx *log.Context, h HandlerEnvironment, report *RunCommandInstanceVi
 	}
 
 	// Report the output streams to blobs
-	outputFilePosition, err = reportOutputToBlob(stdoutF, outputBlobRef, outputFilePosition)
-	errorFilePosition, err = reportOutputToBlob(stderrF, errorBlobRef, errorFilePosition)
+	outputFilePosition, err = appendToBlob(stdoutF, outputBlobSASRef, outputBlobAppendClient, outputFilePosition, ctx)
+	errorFilePosition, err = appendToBlob(stderrF, errorBlobSASRef, errorBlobAppendClient, errorFilePosition, ctx)
 
 	return stdoutTail, stderrTail, runErr
 }
 
-// reportOutputToBlob save a file (from seeking position to the end of the file) to append blob. Returns the new position (end of the file)
-func reportOutputToBlob(sourceFilePath string, outputBlobRef *storage.Blob, outputFilePosition int64) (int64, error) {
-	var err error = nil
-	if outputBlobRef != nil {
+// appendToBlob saves a file (from seeking position to the end of the file) to AppendBlob. Returns the new position (end of the file)
+func appendToBlob(sourceFilePath string, appendBlobRef *storage.Blob, appendBlobClient *appendblob.Client, outputFilePosition int64, ctx *log.Context) (int64, error) {
+	var err error
+	var newOutput []byte
+	if appendBlobRef != nil || appendBlobClient != nil {
 		// Save to blob
-		newOutput, err := getFileFromPosition(sourceFilePath, outputFilePosition)
+		newOutput, err = getFileFromPosition(sourceFilePath, outputFilePosition)
 		if err == nil {
 			newOutputSize := len(newOutput)
 			if newOutputSize > 0 {
-				err = outputBlobRef.AppendBlock(newOutput, nil)
+				if appendBlobRef != nil {
+					err = appendBlobRef.AppendBlock(newOutput, nil)
+				} else if appendBlobClient != nil {
+					ctx.Log("message", fmt.Sprintf("inside appendBlobClient. Output is '%s'", newOutput))
```

Comment:
Yes, it is because we do not know what users will output. Also, I believe logs are meant for diagnostic purposes, to know the extension execution is working or not. For users, they have option streaming output to the blob and debugging it